lib/osc-loader.js

```diff
@@ -0,0 +1,93 @@
+'use babel'
+
+import * as path from 'path';
+import { spawn } from 'child_process';
+import { EventEmitter } from 'events';
+
+const oscPortProperty = 'tidalcycles.oscEvalPort';
+const oscAddressProperty = 'tidalcycles.oscEvalAddress';
+
+export default class OscLoader extends EventEmitter {
+
+    constructor(tidalRepl) {
+        super();
+        console.log("listening on port", atom.config.get(oscPortProperty));
+        this.port = atom.config.get(oscPortProperty);
+        this.server = spawn(
+            'node',
+            [path.resolve(__dirname, 'osc-server.js'), this.port.toString()],
+            {
+                cwd: path.resolve(__dirname, '..'),
+            },
+        );
+        this.server.stdout.on('data', this.stdout);
+        this.server.stderr.on('data', this.stderr);
+        this.server.on('exit', this.exit);
+
+        this.on(atom.config.get(oscAddressProperty), (args) => {tidalRepl.eval(args[0], false);});
+    }
+
+
+    setPort(port) {
```

Comment:
This method is not used, I think the port can be set only on plugin reboot

Reply:
I totally agree with that. The port should only comes from the plugin properties. 
Done